python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -368,24 +353,39 @@ def _(expr: Agg, child_ir: IR) -> MutableMapping[Any, Any]:
     child_key = get_key_name(child)
     child_dsk = generate_expr_tasks(child, child_ir)
 
+    # Check for simple case
+    # TODO: Avoid generating entire child_dsk graph?
+    if expr.name in ("first", "last"):
+        if expr.name == "last":
+            index = npartitions_in - 1
+        else:
+            index = 0
```

Comment:
Note: this only works if the last (resp. first) partition is non-empty.

Reply:
True. I guess we can do a proper reduction to make sure we capture the first/last item.